lib/boxen/preflight/creds.rb

```diff
@@ -105,4 +122,28 @@ def fetch_from_env(thing)
     warn "Oh, looks like you've provided your #{thing} as environmental variable..."
     found
   end
+
+  def fingerprint
+    @fingerprint ||= begin
+      # See Apple technical note TN1103, "Uniquely Identifying a Macintosh
+      # Computer."
+      serial_number_match_data = IO.popen(
+        ["ioreg", "-c", "IOPlatformExpertDevice", "-d", "2"]
+      ).read.match(/"IOPlatformSerialNumber" = "([[:alnum:]]+)"/)
+      if serial_number_match_data
+        # The fingerprint must be unique across all personal access tokens for a
+        # given user. We prefix the serial number with the application name to
+        # differeniate between any other personal access token that uses the
+        # Mac serial number for the fingerprint.
+        Digest::SHA256.hexdigest("Boxen: #{serial_number_match_data[1]}")
```

Comment:
The comment above explains the reasoning here. For normal OAuth apps we never have to worry about this, since the `fingerprint` must only be unique for a given client_id. But, with PATs there is no unique/real client_id, so `fingerprint` must be unique across all PATs.